src/containers/namespace-detail/namespace-detail.tsx

```diff
@@ -224,6 +231,24 @@ export class NamespaceDetail extends React.Component<IProps, IState> {
           collection={updateCollection}
           namespace={namespace.name}
         />
+        <DeleteCollectionModal
+          deleteCollection={deleteCollection}
+          isDeletionPending={isDeletionPending}
+          confirmDelete={confirmDelete}
+          collectionVersion={null}
+          cancelAction={() => this.setState({ deleteCollection: null })}
+          deleteAction={() =>
+            this.setState({ isDeletionPending: true }, () => {
+              deleteCollectionUtils.deleteCollection(
+                this,
```

Comment:
can we be a bit more specific instead of passing `this`?

```suggestion
                {
                  state: this.state,
                  setState: this.setState,
                  load: this.load,
                },
```

Reply:
But this will cause to redo everything else Im doing now in this and Namespace PR (https://github.com/ansible/ansible-hub-ui/pull/2115). The code can remain the same, but I need to replace all this references by the object and this will be no much different than props. I suggest we disscuss this change, maybe in thursday in office, if we will meet there.

Reply:
how would it cause you to redo everything? You're still passing the same methods, just being explicit about which they are.

Passing whole class component instances to other functions is not maintainable, we wouldn't know which functions are being used, we couldn't rename/transform stuff when needed, and there's no `this` in functional components anyway.

Passing `state`, `setState()` and `reload()` seems like a reasonable interface that could be usable from any stateful component (though eventually we'll need to use types to validate the passed in state for compatibility).

Reply:
State and setState seems uselless to me, because every class component has state and setState and it really is not checking anything - we will access further variables of state and setState without type checking. In functional component, it makes sense, because it will pass its own state and setState for every variable, so type checking will do its work. The "load: this.load" can work, but it will cause problems because this will not work anymore in the function, I would have to use lambda and the whole code will become very talkative.  I now really understand why they introduced functional components :-) So what now? Lets pass every state variable and functions as lambda call and use only setState as "magic" dirty access to class state in helper functions? Or pass everything and not use direct state access in helpers at all? I agree that we need mantainability, even if that meens talkative code.

Reply:
Well, this way, all the functions can accept the same interface (state, setState, reload), but do their own type-checking by imposing type constrains on the state param.

That or we can try passing individual getter/setter pairs, but that does seem too talkative indeed, that's what interfaces are for.

(And yeah, you may need `load: () => this.load()` (and similar for setState) for this to work.)